packages/headless/src/controllers/triggers/headless-redirection-trigger.ts

```diff
@@ -0,0 +1,99 @@
+import {Schema, Value} from '@coveo/bueno';
+import {Engine} from '../../app/headless-engine';
+import {
+  ConfigurationSection,
+  RedirectionSection,
+} from '../../state/state-sections';
+import {configuration, redirection} from '../../app/reducers';
+import {buildController, Controller} from '../controller/headless-controller';
+import {loadReducerError} from '../../utils/errors';
+import {validateOptions} from '../../utils/validate-payload';
+import {logTriggerRedirect} from '../../features/redirection/redirection-analytics-actions';
+
+export interface RedirectionTriggerOptions {
+  /**
+   * The function used to handle whenever the `Redirection` controller's state's `redirectTo` value changes.
+   */
+  onRedirect(): void;
+}
+
+export interface RedirectionTriggerProps {
+  /**
+   * The options for the `Redirection` controller.
+   */
+  options?: RedirectionTriggerOptions;
+}
+
+const optionsSchema = new Schema({
+  onRedirect: new Value({required: true}),
+});
+
+/**
+ * The `Redirection` controller handles redirection actions.
+ */
+export interface RedirectionTrigger extends Controller {
+  /**
+   * the state of the Redirection controller.
+   */
+  state: RedirectionTriggerState;
+}
+
+/**
+ * A scoped and simplified part of the headless state that is relevant to the `Redirection` controller.
+ */
+export interface RedirectionTriggerState {
+  /**
+   * The url used for the redirection.
+   */
+  redirectTo: string;
+}
+
+/**
+ * Creates a `Redirection` controller instance.
+ *
+ * @param engine - The headless engine.
+ * @param props - The configurable `Redirection` properties.
+ * @returns A `Redirection` controller instance.
+ * */
+export function buildRedirection(
+  engine: Engine<object>,
+  props: RedirectionTriggerProps = {}
+): RedirectionTrigger {
+  if (!loadRedirectionReducers(engine)) {
+    throw loadReducerError;
+  }
+
+  const controller = buildController(engine);
+  const {dispatch} = engine;
+
+  const options = validateOptions(
+    engine,
+    optionsSchema,
+    props.options,
+    'buildRedirectionTrigger'
+  ) as Required<RedirectionTriggerOptions>;
+
+  const redirectTo = engine.state.redirection.redirectTo!;
```

Comment:
Here, you only access the state value once, when the `buildRedirection` method is called, but it's not updated each time the engine state's changes. Here a few cues:

- this line could be a method, returning the state, so every time it's called it's the current state
- when you make the comparison, at line 78, it's with that original value, but it's never compared further, consider using the `engine.subscribe` method, you can also override the controller's own subscribe method, like it's done for `headless-url-manager.ts`